File: donuts/donut.go
             package main
         import("fmt";."math"
       "time");func main(){A,B:=
     0.,0.;b:=make([]byte,1760);for{
   var z[1760]float64;for i:=range b{
  b[i]=32};for j:=0.;6.28>j;j+=0.07{for
 i:=0.;6.28>i;i+=0.02{c,d,e,f,g,l,m,n:=
 Sin(i),Cos(j),Sin(A),Sin(j),Cos(A),Cos(
 i),Cos(B),Sin(B)      ;h:=d+2;D,t:=1/(c*
h*e+f*g+5),c*h*g        -f*e;N,x,y:=int(8*
((f*e-c*d*g)*m             -c*d*e-f*g-l*d*
n)),int(40+30*D*         (l*h*m-t*n)),int(
 12+15*D*(l*h*n+t      *m));o:=x+80*y;if//
 22>y&&y>0&&x>0&&80>x&&D>z[o]{z[o]=D;p:=0
 if N>0{p=N}else{p=0};b[o]=(".,-~:;=!*#"+
  "$@")[p]}}};for k:=range b{if k%80==0{
   b[k]=10}};fmt.Println("\x1b[H",//--.
     string(b));A+=0.04;B+=0.02;time.
       Sleep(time.Second/15)}}/**
         ;;==#####@@@@@@#####=;
             ;==$##@@#$=;*/

File: donuts/donut.pl
               use Math::Trig; use 
           Time::HiRes qw(usleep);my$y
         =1760;my $a=0.0; my $b=0.0; sub
       join_str{my($s, @a)=@_;my$r=""; for
     my$i(0..$y-1){$r.=$a[$i];$r.=$s if$i<$y
    -1}return $r}for(; ;){my@s=("")x$y;my@t=(
   .0)x$y;$a+=.05;$b+=.07;my $o=cos($a);my $e=
  sin($a);my$n=cos($b);my$c=sin($b);for my $i(0
 ..$y-1){$s[$i]=($i           %80==79)?"\n":" ";
 $t[$i]=0}for(my$i             =0;$i<6.28; $i +=
 .07){my$r=cos($i               );my$w= sin($i);
 for(my$j=0; $j <               6.28;$j+=.02){my
 $l=sin($j);my$f=               cos($j);my$g=$r+
 2;my$q=1/($l*$g *             $e+$w*$o+5);my$d=
 $l*$g*$o-$w*$e; my           $m= int(40+30*$q*(
  $f*$g*$n-$d*$c));my$v=int(12+15*$q*($f*$g*$c+
   $d*$n));my$x=$m+80*$v;my$h=int(8*(($w*$e-$l
    *$r*$o)*$n-$l*$r*$e-$w*$o-$f*$r*$c));if (
     $v<22&&$v>=0&&$m>=0&&$m<79&&$q>$t[$x]){
       $t[$x]=$q;$s[$x]=substr((".,-~:;=!".
         "*#\$@"),$h>0?$h:0,1)}}}print("
           \x1b[J\x1b[H".join_str("" ,
               @s));usleep(50000)}